rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -1645,4 +1819,140 @@ private static Stream<Arguments> malFormedCoinbaseTxAndExpectedWitnessProvider()
             );
         }
     }
+
+    private void assertSegwitScriptSigContainsHashedRedeemScript(Script segwitScriptSig, Script redeemScript) {
+        List<ScriptChunk> chunks = segwitScriptSig.getChunks();
+        assertEquals(1, chunks.size());
+
+        ScriptChunk chunk = chunks.get(0);
+        assertEquals(34, chunk.opcode); // OP_PUSHBYTES_34, 32 bytes from the redeem script hash + 2 for OP_0 and OP_PUSHBYTES_32
+
+        byte[] segwitScriptSigProgram = segwitScriptSig.getProgram();
+        assertEquals(35, segwitScriptSigProgram.length); // OP_PUSHBYTES_34 + the 34 bytes
+
+        // Check the first byte is OP_PUSHBYTES_34
+        assertEquals(34, segwitScriptSigProgram[0]);
+
+        // Check the second byte is OP_0
+        assertEquals(0, segwitScriptSigProgram[1]); // OP_0
```

Comment:
```suggestion
        assertEquals(0, segwitScriptSigProgram[1]);
```
prob no need for that comment since you said it above

Reply:
true